hydra-http.c

```diff
@@ -34,7 +45,7 @@ int32_t start_http(int32_t s, char *ip, int32_t port, unsigned char options, cha
 
   buffer_size = strlen(header) + 500;
   if (!(buffer = malloc(buffer_size))) {
-    free(header);
+    freeM(header);
```

Comment:
this - and all other - would need to be `&header` instead of `header` (because the function is void **ptr)

Reply:
```suggestion
    void freeMemory(void **ptr) {
  if (ptr != NULL && *ptr != NULL) {
    free(*ptr);
    *ptr = NULL;
  }
}
```

Maybe I think that's correct here, with the processing inside the function body

Reply:
have you tested it? it should crash :) 
just try `char *foo = malloc(100); freeM(foo);`

Reply:
If I use &header, it will report an error
```suggestion
    freeM(&
<img width="612" alt="20210707-231705@2x" src="https://user-images.githubusercontent.com/6275408/124785501-7ad8d680-df79-11eb-9df5-3eb043b97965.png">
header);
```

Reply:
> have you tested it? it should crash :)
> just try `char *foo = malloc(100); freeM(foo);`

yes, I tried it and it didn't crash

Reply:
Maybe i think you overlooked it
```c
void freeMemory(void **ptr) {
  if (ptr != NULL && *ptr != NULL) {
    free(*ptr);
    *ptr = NULL;
  }
}

#define freeM(p) freeMemory((void **)(&(p)))
```

Reply:
yeah I had overlooked that :)